0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+The message body should be a JSON object with one field, `fulfillment`, which contains a [Crypto-Condition Fulfillment][] in string format.
+
+#### Response Format
+
+A successful result uses the HTTP response code **201 Created**. The message body of the response is a JSON object containing the submitted fulfillment in the `fulfillment` field. A ledger MUST return a successful response if and only if the transfer was executed as a result of this request.
+
+If you
+
+#### Example
+
+Request:
+
+```json
+PUT /transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204/fulfillment
+Content-Type: application/json
+
+{
+  "fulfillment": "cf:0:_v8"
+}
+```
+
+Response:
+
+```json
+HTTP/1.1 200 OK
+
+{
+  "fulfillment": "cf:0:_v8"
+}
+```
+
+#### Errors
+
+- [NotFoundError][] - The transfer does not exist.
+- [UnmetConditionError][] - The fulfillment does not match the condition.
+- [TransferNotConditionalError][] - The transfer had no condition to fulfill.
+- [TransferStateError][] - The transfer was not in the `prepared` state when the request was received. This occurs if the transfer has already been executed, rejected, or expired.
+- [UnprocessableEntityError][] - The request is formatted properly but contains an otherwise-unspecified semantic problem.
+- [InvalidBodyError][] - The request body was not properly-formatted JSON.
+
+
+### Reject Transfer
+[Reject Transfer]: #reject-transfer
+
+Reject a prepared transfer. A transfer can be rejected if and only if that transfer is in the `prepared` state. Doing so transitions the transfer to the `rejected` state.
+
+**Authorization:** The `credit_account` of a transfer MUST be able to reject the transfer. Unauthorized users and the `debit_account` MUST NOT be able to reject the transfer. Other account owners MUST NOT be able to reject the transfer unless those accounts are explicitly authorized to do so by the `credit_account` and those accounts are not the `debit_account`.
+
+#### Request Format
+
+PUT to the `transfer_rejection` path of the transfer you want to execute, with a JSON object in the message body containing a rejection reason.
+
+**Metadata URL Name:** `transfer_rejection`. Suggested generic path: `/transfers/{client_id}/rejection`
+
+##### Body Parameters
+
+The message body should be a JSON object with one top-level field:
+
+| Field              | Value  | Description                                    |
+|:-------------------|:-------|:-----------------------------------------------|
+| `rejection_reason` | String | The reason for rejecting the transfer. This is intended to be a machine-readable identifier. |
+
+#### Response Format
+
+A successful result uses the HTTP response code **200 OK** and contains a JSON object containing the updated [Transfer resource][] containing the rejection reason.
+
+#### Example
+
+Request:
+
+```json
+PUT /transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204/rejection
+Content-Type: application/json
+
+{
+  "rejection_reason": "BlacklistedSender"
+}
+```
+
+Response:
```

Comment:
This still seems odd to me that you send just the `rejection_reason` and get back the whole transfer. It's also inconsistent with how the fulfillments work because according to this logic you would want to see the whole transfer for that too.

I think it's probably better to just return the `rejection_reason` and rely on notifications to give you updates about the transfer state. In that vein, we should make a note about whether an `HTTP 200` indicates the transfer was definitely rejected